--- php-7.2.5-origin/main/reentrancy.c	2018-04-24 23:10:05.000000000 +0800
+++ php-7.2.5/main/reentrancy.c	2018-05-29 16:05:58.228888471 +0800
@@ -128,6 +128,7 @@
 PHPAPI int php_readdir_r(DIR *dirp, struct dirent *entry,
 		struct dirent **result)
 {
+	return 0;
 #if defined(HAVE_OLD_READDIR_R)
 	int ret = 0;

@@ -136,7 +137,7 @@
 	   (HPUX returns 0 on success whereas Solaris returns non-zero)
 	 */
 	entry->d_name[0] = '\0';
-	readdir_r(dirp, entry);
+	//readdir_r(dirp, entry);

 	if (entry->d_name[0] == '\0') {
 		*result = NULL;
--- php-7.2.5-origin/main/streams/plain_wrapper.c	2018-04-24 23:10:05.000000000 +0800
+++ php-7.2.5/main/streams/plain_wrapper.c	2018-05-29 16:05:58.228888471 +0800
@@ -1364,7 +1364,7 @@
 				fclose(file);
 			}

-			ret = VCWD_UTIME(url, newtime);
+			//ret = VCWD_UTIME(url, newtime);
 			break;
 #ifndef PHP_WIN32
 		case PHP_STREAM_META_OWNER_NAME:
